txclib/utils.py

```diff
@@ -143,13 +143,25 @@ def _prepare_url_request(host, username, password):
 
     [1]: http://urllib3.readthedocs.io/en/latest/reference/#urllib3.poolmanager.ProxyManager  # noqa
     """
+    def get_proxy_url(env_proxy, enforce_scheme):
+        parsed_proxy = urllib3.util.url.parse_url(env_proxy)
+        return urllib3.util.url.Url(
+            scheme=enforce_scheme,
+            auth=parsed_proxy.auth,
+            host=parsed_proxy.host,
+            port=parsed_proxy.port,
+            path=parsed_proxy.path,
+            query=parsed_proxy.query,
+            fragment=parsed_proxy.fragment
+        )
+
```

Comment:
I would change `enforce_scheme` to just `scheme` (plus the verb "enforce" suits a function name rather than a variable/argument name).

Reply:
I assume `Url` is immutable and that's why you are creating a new object instead of changing the existing one?

Reply:
Correct:
```
>>> import urllib3.util.url
>>> parsed_proxy = urllib3.util.url.parse_url('http://www.in.gr')
>>> parsed_proxy.scheme = 'https'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: can't set attribute
>>>
```